paddle/fluid/imperative/hooks.h

```diff
@@ -18,100 +18,67 @@
 #include <memory>
 #include <utility>
 #include <vector>
-
-#include "paddle/fluid/imperative/type_defs.h"
-#include "paddle/fluid/platform/macros.h"
-
 namespace paddle {
 namespace imperative {
 
 class VariableWrapper;
 
-/** [ Basic hook classes ]
- * s
- * @brief OpBasePreHook is executed before the grad OpBase is executed,
+/** [ Const VariableWrapper Hook: Pre hook functor of OpBase ]
+ *
+ * @brief This hook functor is executed before the grad OpBase is executed,
  *        taking the input of the current grad OpBase as input, and
  *        executing python hooks (user-defined) or C++ hooks (developer-defined)
  *        to achieve the purpose of custom operations on the interior VarBase
  *        gradient.
  *
- * @note  OpBasePreHook will not change the input gradient VarBase.
+ * @note  This hook functor will not change the input gradient VarBase.
  *
  * @note  [Why need to be OpBase `PreHook`, why not `PostHook`?]
  *
- *        If set OpBase post hook, when the op executed end, the op's output
- *        gradient may not be the final state, because it may need other op's
- *        gradient output to accumulated to it. But before op can be executed,
- *        the gradient output must have been accumulated to final value.
+ *        1. We expect If set OpBase post hook, when the op executed end, the
+ *        op's output gradient may not be the final state, because it may need
+ *        other op's gradient output to accumulated to it. But before op can
+ *        be executed, the gradient output must have been accumulated to final
+ *        value.
+ *        2. We don’t want the hook to change its input Tensor value, so now
+ *        we can't call all hooks in GradAccumulator.
  *
  * @note  [Why only can be used for interior VarBase?]
  *
  *        Because the leaf VarBase's GradVarBase has no GradOpNode, so leaf
  *        GradVarBase has no next OpBase to executed, so if need to deal with
- *        the leaf GradVarBase, cannot use OpBasePreHook. For this case, we
- *        deal with by GradAccumulatorPostHook.
+ *        the leaf GradVarBase, cannot use this hook functor. For this case, we
+ *        deal with by other inplace hook method.
  */
-class OpBasePreHook {
+class VariableWrapperHook {
```

Comment:
how about make a abstract class of Hook to seal different kinds of hooks

Reply:
I have tried, that is not a good idea